src/vm/fieldmarshaler.cpp

```diff
@@ -1254,9 +1254,11 @@ BOOL IsFixedBuffer(mdFieldDef field, IMDInternalImport* pInternalImport, LONG* p
     BYTE* pData;
     LONG cData;
 
-    HRESULT hr = pInternalImport->GetCustomAttributeByName(field, g_FixedBufferAttribute, (const VOID **)(&pData), (ULONG *)&cData);
+    HRESULT hr;
+
+    IfFailGo(pInternalImport->GetCustomAttributeByName(field, g_FixedBufferAttribute, (const VOID **)(&pData), (ULONG *)&cData));
 
-    if (SUCCEEDED(hr) && cData != 0)
+    if (hr == S_OK && cData != 0)
```

Comment:
Sometimes `GetCustomAttributeByName` returns a positive (non-failure) status code and sets `cData` to a non-zero value, but the type doesn't have the attribute. So, this would cause an AV in `ParseKnownCaArgs`. By checking specifically if `hr == S_OK`, we only try to parse if the attribute exists.

Reply:
> Sometimes GetCustomAttributeByName returns a positive (non-failure) status code and sets cData to a non-zero value, but the type doesn't have the attribute.

just curious when this happens(GetCustomAttributeByName returns a positive (non-failure) status code and sets cData to a non-zero value)? this looks weird for me.

Reply:
in my mind, this code should be able to transform into 

if(FAILED(pInternalImport->GetCustomAttributeByName(field, g_FixedBufferAttribute, (const VOID **)(&pData), (ULONG *)&cData))
    return false;
if(cData != 0)
....

Reply:
It was happening on the `KeyValuePair<TKey, TValue>` type if I remember correctly. Not sure why it was happening, but it would cause crossgen and the JIT to AV.

Reply:
I'll make that change!

Reply:
That change still causes an AV. It needs to compare to S_OK.